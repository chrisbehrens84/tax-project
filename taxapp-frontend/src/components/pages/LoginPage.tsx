import { Form, Label, TextInput, Button } from "@trussworks/react-uswds";
import {useNavigate} from 'react-router-dom'

export default function LoginPage(){

    const navigate = useNavigate();

    function loginFormSubmit(event : any){
        event.preventDefault();
    }

    function signUpButton(event : any){
        event.preventDefault();
        navigate('/signup');
    }



    return(
        <>
            <div style={{marginLeft : '4rem'}}>
                <Form onSubmit={loginFormSubmit}>
                    <Label htmlFor="emailInput">Email:</Label>
                    <TextInput id='emailInput' name='emailInput' type='text'/>
                    <Label htmlFor="passwordInput">Password:</Label>
                    <TextInput id='passwordInput' name='passwordInput' type='text'/>
                    <Button type="submit">Log In</Button>
                    <Button type="button" onClick={signUpButton}>Sign up</Button>
                </Form>
            </div>
        </>
    )
}