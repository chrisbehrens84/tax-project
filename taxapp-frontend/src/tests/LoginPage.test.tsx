//import React from "react";
import renderer from "react-test-renderer"
import LoginPage from "../components/pages/LoginPage";

const mockUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
   ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockUsedNavigate,
}));


it('Login Page rendering', () =>{

    //Creates the json data for HTML tree
    const tree : any = renderer.create(<LoginPage/>).toJSON();

    //Compares to existing snapshot OR generates new one
    expect(tree).toMatchSnapshot();
    
});