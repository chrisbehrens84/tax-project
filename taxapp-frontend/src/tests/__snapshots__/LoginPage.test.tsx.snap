// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Home Page rendering 1`] = `
<div
  style={
    {
      "marginLeft": "4rem",
    }
  }
>
  <form
    className="usa-form"
    data-testid="form"
    onSubmit={[Function]}
  >
    <label
      className="usa-label"
      data-testid="label"
      htmlFor="emailInput"
    >
      Email:
    </label>
    <input
      className="usa-input"
      data-testid="textInput"
      id="emailInput"
      name="emailInput"
      type="text"
    />
    <label
      className="usa-label"
      data-testid="label"
      htmlFor="passwordInput"
    >
      Password:
    </label>
    <input
      className="usa-input"
      data-testid="textInput"
      id="passwordInput"
      name="passwordInput"
      type="text"
    />
    <button
      className="usa-button"
      data-testid="button"
      type="submit"
    >
      Log In
    </button>
    <button
      className="usa-button"
      data-testid="button"
      onClick={[Function]}
      type="button"
    >
      Sign up
    </button>
  </form>
</div>
`;
